### PR TITLE
Pipeline sync looks up jobs in screwdriver.yaml and creates them all

### PR DESCRIPTION
**Pull request by petey**
Use scm plugin to fetch screwdriver.yaml, parse contents, and create all jobs listed.
- Will add update & delete in following PR(s)

### CODE REVIEW

**Comment by nkatzman** [comment]
👍  looks good

File: lib/pipeline.js [hunk 1]
```diff
@@ -31,13 +33,41 @@ class PipelineModel extends BaseModel {
...
+                    // add a method to create job
+                    jobsToCreate.push(factory.create(jobConfig));
+                });
+
+                return jobsToCreate;
+            })
+            // Wait until all job.create promises have resolved
+            .then(jobs => Promise.all(jobs))
+            // return the "main" job
+            .then(jobs => jobs.filter(j => j.name === 'main')[0]);
```
Comment (stjohnjohnson) [inline lib/pipeline.js]: I wish we just returned the pipeline, we don't need the jobs.

**Comment by stjohnjohnson** [comment]
LGTM 👍  
wat

File: lib/pipeline.js (hunk 1 above)
Comment (d2lam) [inline lib/pipeline.js]: we could

**Pull request merged**